v1/ast/builtins.go

```diff
@@ -2063,6 +2065,32 @@ var YAMLIsValid = &Builtin{
 	CanSkipBctx: true,
 }
 
+var PurlIsValid = &Builtin{
+	Name:        "purl.is_valid",
+	Description: "Validates that the input is a valid Package URL (PURL).",
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("purl", types.S).Description("Package URL string to validate"),
+		),
+		types.Named("result", types.B).Description("`true` if `purl` is a valid Package URL; `false` otherwise"),
+	),
+	Categories:  encoding,
+	CanSkipBctx: true,
+}
+
+var PurlParse = &Builtin{
+	Name:        "purl.parse",
+	Description: "Parses a Package URL (PURL) string into its components.",
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("purl", types.S).Description("Package URL string to parse"),
+		),
+		types.Named("result", types.NewObject(nil, types.NewDynamicProperty(types.S, types.A))).Description("object containing PURL components: type, namespace, name, version, qualifiers, subpath"),
```

Comment:
The Conforma (prev Enterprise Contract) project's implementation defined each of the expected attributes as static properties: https://github.com/conforma/cli/blob/d0c907ed9c9df8d5b6c24df2bcfd27e8d2ec78ad/internal/rego/purl/purl.go#L70-L90

I found it helpful when authoring rules as the typos were easily caught by opa.

Reply:
Thank you!